### ISSUE TITLE
won't build in ubuntu 24.10

### ISSUE THREAD

**Issue by mmetc** [wasilibs/go-re2#152]
Hi,

I have a build error with the new c++ toolchain in 24.10

`github.com/wasilibs/go-re2/internal/cre2: invalid flag in pkg-config --libs: -Wl,--push-state,--as-needed`

As a hack, I am explicitly providing the build flags, but older versions of re2 don't require abseil to build, so it's not a proper fix for us (crowdsec).

```
diff --git a/internal/cre2/cre2_re2_cgo.go b/internal/cre2/cre2_re2_cgo.go
index 8f0e5dd..cef0d78 100644
--- a/internal/cre2/cre2_re2_cgo.go
+++ b/internal/cre2/cre2_re2_cgo.go
@@ -3,7 +3,7 @@
 package cre2
 
 /*
-#cgo pkg-config: re2
 #cgo CXXFLAGS: -std=c++17
+#cgo LDFLAGS: -lre2 -labsl_flags_internal -labsl_flags_marshalling -labsl_log_severity -labsl_flags_reflection -labsl_flags_private_handle_accessor -labsl_flags_commandlineflag -labsl_flags_commandlineflag_internal -labsl_flags_config -labsl_flags_program_name -labsl_raw_logging_internal -labsl_raw_logging_internal -labsl_cord -labsl_cordz_info -labsl_cord_internal -labsl_cordz_functions -labsl_cordz_handle -labsl_crc_cord_state -labsl_crc32c -labsl_crc_internal -labsl_crc_cpu_detect -labsl_raw_hash_set -labsl_hash -labsl_city -labsl_bad_variant_access -labsl_low_level_hash -labsl_hashtablez_sampler -labsl_exponential_biased -labsl_raw_logging_internal -labsl_bad_optional_access -labsl_str_format_internal -labsl_synchronization -labsl_graphcycles_internal -labsl_kernel_timeout_internal -labsl_stacktrace -labsl_symbolize -labsl_debugging_internal -labsl_demangle_internal -labsl_malloc_internal -labsl_time -labsl_civil_time -labsl_strings -labsl_string_view -labsl_strings_internal -labsl_base -latomic -lrt -labsl_spinlock_wait -labsl_int128 -labsl_throw_delegate -labsl_raw_logging_internal -labsl_log_severity -labsl_time_zone 
 */
 import "C"
```

I am using the flags from "pkg-config --libs-only-l re2", but could not find a way to call it automatically.

Any idea? Am I missing something? Thanks

**Comment by anuraaga** [comment]
Hi @mmetc - as far as I know, cgo is just invoking pkg-config which is mostly reading from the pc file installed by the c library. So this seems like an issue with re2's installation rather than this repo, possibly because you are using an older version. I know the absl dep makes compiling newer versions more challenging (I still haven't found a clean way for the wasm cross compile...) but maybe you can try it? Alternatively, any reason to avoid libre2-dev? It's what we use in the Ubuntu CI here.

**Comment by anuraaga** [comment]
Oh I realized the same error happens with ubuntu's built-in libre2-dev so it's not an issue specific to using an older version. I filed an issue in Go repo as I think this needs to be fixed on the cgo side

https://github.com/golang/go/issues/70023

Unfortunately until it's fixed, I don't think there is much we can do on this side since we rely on the pkg-config integration to work properly, and I think it's ok for us to have some delay on supporting 24.10 given it's not an LTS release.

As an aside, I noticed the change to use libre2 by default from last year https://github.com/crowdsecurity/crowdsec/pull/2295

Since [v1.5.0](https://github.com/wasilibs/go-re2/releases/tag/v1.5.0) a lot of the memory usage issues with the Wasm backend have been solved and the tradeoffs vs cgo have been reduced significantly. It still may not be enough for a high-intensity use case like crowdsec, but it could be interesting in giving wasm a try again as default to see if it's viable. Anyways, I'm hoping the cgo issue gets solved to bring compatibility to ubuntu 24.10. Thanks for the report it was helpful knowing the situation on this.
